### PR TITLE
Update Header to be more responsive

### PR DESCRIPTION
**Pull request by joeytepp**
## Proposed Change

Fixes: #117

Adds the coveted _s_ _t_ _i_ _c_ _k_ _y_ `Header` to the QHacks website (surprisingly not actually `position: sticky`).

### How did you do this?
Added a `Header` component in which the `Menu` is housed. This will only appear when scrolling upwards or at the top of the page. Also added the `react-scrollspy` package so that the menu is aware of which section you are scrolling over. Also added the `MobileMenu` component to match the mobile mocks, and moved the `BriefSummary` and `SellingPoints` components into a single `About` component to better match the navigation structure of the website

### Why are you choosing this approach?
To keep things modular and match the mocks

### Any open questions you want to ask code reviewers?
No

### List of changes:

  - Add `Header` component
  - Add `MobileMenu` component
  - Move `BriefSummary` and `SellingPoints` into `About` component

## Pull Request Checklist:

  - [x] My submission passes all tests.
  - [x] I have lint my code locally prior to submission.
  - [x] I have written new tests for my core changes (where applicable).
  - [x] I have referenced all useful information (issues, tasks, etc) in the pull request.

### CODE REVIEW

**Comment by RobertWSaunders** [comment]
Deploy preview for *qhacks-website* failed.

Built with commit 67307a701bb9657917972825321f87af520fea74

https://app.netlify.com/sites/qhacks-website/deploys/5bca0bc23813f07262ec9f3d

File: package.json [hunk 1]
```diff
@@ -40,6 +40,7 @@
     "react": "^16.5.0",
     "react-dom": "^16.5.0",
     "react-helmet": "^5.2.0",
+    "react-scrollspy": "^3.3.5",
```
Comment (RobertWSaunders) [inline package.json]: Nice! 🔥

File: src/assets/img/applications.svg [hunk 2]
```diff
@@ -1,9 +1,9 @@
 <svg xmlns="http://www.w3.org/2000/svg" width="43" height="32" viewBox="0 0 43 32">
-    <g fill="none" fill-rule="nonzero">
```
Comment (RobertWSaunders) [inline src/assets/img/applications.svg]: Nice change here to make things consistent!

File: src/assets/img/hamburger.svg [hunk 3]
```diff
@@ -0,0 +1,3 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="32" height="24" viewBox="0 0 32 24">
+   <path fill="#00205B" fill-rule="nonzero" d="M2 0h28a2 2 0 1 1 0 4H2a2 2 0 1 1 0-4zm0 10h28a2 2 0 1 1 0 4H2a2 2 0 1 1 0-4zm0 10h28a2 2 0 1 1 0 4H2a2 2 0 1 1 0-4z"/>
```
Comment (RobertWSaunders) [inline src/assets/img/hamburger.svg]: This looks like three spaces!

File: src/components/About.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import React from "react";
+import BriefSummary from "./BriefSummary";
+import SellingPoints from "./SellingPoints";
+
+const About = () => (
```
Comment (RobertWSaunders) [inline src/components/About.js]: I like this change, it's nice to follow the menu! 👍

File: src/components/Header.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+      ) {
+        this.setState({ isHeaderVisible, isLogoVisible });
+      }
+      prevScrollY = window.scrollY;
+    };
+  }
+
+  onMobileMenuClicked = () => {
+    this.setState({ isMobileMenuVisible: !this.state.isMobileMenuVisible });
+  };
```
Comment (RobertWSaunders) [inline src/components/Header.js]: Can we add a new line here? Nitpick.

File: src/components/Header.js [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    return (
+      <div>
+        <div
+          css={{
+            width: "100vw",
+            position: "fixed",
+            top: this.state.isHeaderVisible ? "0px" : "-100px",
+            transition: "0.5s",
+            backgroundColor: "#ffffff",
+            opacity: this.state.isMobileMenuVisible ? "1.0" : "0.8",
```
Comment (RobertWSaunders) [inline src/components/Header.js]: When scrolled to the top can we make the opacity zero on desktop? See the image below, I don't like it interferes with the elements in the graphic, thoughts?

screen shot 2018-10-21 at 4 09 48 pm

File: src/components/Landing.js [hunk 7]
```diff
@@ -4,15 +4,15 @@ import word from "../assets/img/wordmark.svg";
 import backgroundCubes from "../assets/img/backgroundCubes.svg";
 import landingImg from "../assets/img/landingImg.png";
 import ActionButton from "./ActionButton";
-import Menu from "./Menu";
+import Header from "./Header";
 
 const breadCrumbs = [
```
Comment (RobertWSaunders) [inline src/components/Landing.js]: is "bread crumbs" the name of these or are "menu items" a better name?

File: src/pages/index.js [hunk 8]
```diff
@@ -50,8 +49,7 @@ export default () => (
     />
     <Landing />
     <MailingListSignup />
-    <BriefSummary />
-    <SellingPoints />
+    <About />
```
Comment (RobertWSaunders) [inline src/pages/index.js]: Nice!

File: src/components/Landing.js [hunk 9]
```diff
@@ -21,17 +21,16 @@ const loginButtonCSS = {
 
 const Landing = () => (
   <div
-    className="landing"
     css={{
       width: "100%",
       height: "100vh",
-      minHeight: "700px",
-      maxHeight: "800px",
+      minHeight: "800px",
```
Comment (RobertWSaunders) [inline src/components/Landing.js]: Why these changes?

File: src/components/Landing.js [hunk 10]
```diff
@@ -42,7 +41,7 @@ const Landing = () => (
         css={{
           paddingLeft: "110px",
           paddingRight: "50px",
-          paddingTop: "42px",
+          paddingTop: "142px",
```
Comment (RobertWSaunders) [inline src/components/Landing.js]: Why this change? It appears like the bottom of the landing page got much taller, see images below:

Before (dev.qhacks.io):

before
0):

After:

after

It looks quite distorted now.

File: src/components/MobileMenu.js [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+              textTransform: "uppercase",
+              ":hover": {
+                color: "#fedb01"
+              }
+            }
+          }}
+        >
+          <div>
+            <img
+              src={logo}
```
Comment (RobertWSaunders) [inline src/components/MobileMenu.js]: Looks like the logo should be a bit bigger, I know you followed the dimensions in Zeplin but it looks off so increasing the logo size would be good.

Zeplin:
screen shot 2018-10-21 at 4 34 30 pm

Yours:
screen shot 2018-10-21 at 4 34 51 pm

File: src/components/Landing.js (hunk 7 above)
Comment (joeytepp) [reply src/components/Landing.js]: `menuItems` is probably a better name, will change it!

File: src/components/Landing.js (hunk 9 above)
Comment (joeytepp) [reply src/components/Landing.js]: The button was overflowing to the section below it but only on mobile, so this should have been in a media query, good catch 👍

Comment (joeytepp) [reply src/components/Landing.js]: This is also going to change when I update the `Landing` component for mobile, but will add a temporary fix for now

File: src/components/Landing.js (hunk 10 above)
Comment (joeytepp) [reply src/components/Landing.js]: Related to the `minWidth` and `maxWidth` changes above, was fixed with the media query

File: src/components/Header.js (hunk 6 above)
Comment (joeytepp) [reply src/components/Header.js]: Sounds good, I'll make it transparent when the logo is present, which is when the scroll height is less than 200px

File: src/components/MobileMenu.js (hunk 11 above)
Comment (joeytepp) [reply src/components/MobileMenu.js]: Seems like getting rid of `width` and `height` in the img did the trick!

**Comment by joeytepp** [comment]
@RobertWSaunders updated!

File: src/components/Header.js (hunk 6 above)
Comment (RobertWSaunders) [reply src/components/Header.js]: I think when the scroll height is zero then it should be transparent otherwise it should be translucent as you had, otherwise I find you get some weird overlapping like below:

screen shot 2018-10-21 at 11 00 41 pm

File: src/components/Landing.js (hunk 10 above)
Comment (RobertWSaunders) [reply src/components/Landing.js]: Looks like the graphic is still moving up, if you have the preview open and dev open in tabs and switch between them they shouldn't change but I still see a change. See below:

Before:
screen shot 2018-10-21 at 10 58 56 pm


After:

screen shot 2018-10-21 at 10 58 50 pm

Comment (RobertWSaunders) [reply src/components/Landing.js]: We can leave this as we still need to resist the entire landing component for alignment issues and mobile responsiveness or we can address now, your call?

Comment (RobertWSaunders) [reply src/components/Landing.js]: The image still seems to be moved up?

File: src/components/Header.js (hunk 6 above)
Comment (RobertWSaunders) [reply src/components/Header.js]: Can we make transparent when scroll height is zero?

File: src/components/Landing.js (hunk 10 above)
Comment (joeytepp) [reply src/components/Landing.js]: Think i'm gonna leave it for now, planning on tackling the `Landing` component next and updating it for mobile so I'll likely be refactoring a lot of the existing CSS

File: src/components/Header.js (hunk 6 above)
Comment (joeytepp) [reply src/components/Header.js]: Sure thing 👌

File: src/components/Landing.js (hunk 10 above)
Comment (RobertWSaunders) [reply src/components/Landing.js]: Sounds good!
